ses-ballerina/src/ses/ses.bal

```diff
@@ -0,0 +1,599 @@
+        if (templateJson is error) {
+            return Error("Error while cloning a template to a JSON map ", templateJson);
+        } else {
+            error? encodeError = self.addTemplateParams(parameters, templateJson);
+            if (encodeError is error) {
+                return Error("Error while encoding template values ", encodeError);
+            }
+            http:Request|error request = self.generatePOSTRequest(endpoint,
+                self.buildPayload(parameters));
+            if (request is http:Request) {
+                var httpResponse = self.clientEp->post(endpoint, request);
+                xml|error response = handleResponse(httpResponse);
+                if (response is error){
+                    return Error("Error while creating the template.", response);
+                }
+            } else {
+                return Error("Error while generating the POST request to create the template.", request);
+            }  
+        }
+    }
+
+    # Deletes an email template.
+    # ```ballerina
+    # ses:Error? result = sesClient->deleteTemplate("MyTemplate");
+    # ```
+    #
+    # + templateName - The name of the template to be deleted
+    # + return - A `ses:Error` if an error occurred while the operation
+    public remote function deleteTemplate(string templateName) returns Error? {
+        string endpoint = DEFAULT_ENDPOINT;
+        string payload;
+        map<string> parameters = {};
+        parameters[PAYLOAD_PARAM_ACTION] = ACTION_DELETE_TEMPLATE;
+        parameters[PAYLOAD_PARAM_VERSION] = SES_VERSION;
+        string|error encodedValue = encoding:encodeUriComponent(templateName, UTF_8);
+        if (encodedValue is error) {
+            return Error("Error while encoding template name ", encodedValue);
+        } else {
+            parameters[TEMPLATE_PARAM_TEMPLATE_NAME] = encodedValue;
+            http:Request|error request = self.generatePOSTRequest(endpoint, self.buildPayload(parameters));
+            if (request is http:Request) {
+                var httpResponse = self.clientEp->post(endpoint, request);
+                xml|error response = handleResponse(httpResponse);
+                if (response is error){
+                    return Error("Error while deleting the template.", response);
+                }
+            } else {
+                return Error("Error while generating the POST request to delete the template.", request);
+            }
+        }
+    }
+
+    private function getTemplateDataString(map<string> templateData) returns string {
+        string templateDataString = "{ ";
+        int i = 0;
+        foreach var [key, value] in templateData.entries() {
+            if (i > 0) {
+                templateDataString = templateDataString + ", ";
+            }
+            templateDataString = string `${templateDataString}\"${key}\":\"${value}\"`;
+            i = i + 1;
+        }
+        templateDataString = templateDataString + " }";
```

Comment:
Is there a possibility of using https://ballerina.io/swan-lake/learn/api-docs/ballerina/lang.string/functions.html#join here ?

Reply:
As it seems `join` is used to join a set of strings separated by a specific `separator` string which is bit different from this implementation.